roles/tobiko/tasks/main.yml

```diff
@@ -0,0 +1,102 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Ensure podman is installed
+  become: true
+  ansible.builtin.package:
+    name: podman
+    state: present
+
+- name: Create tobiko directories
+  ansible.builtin.file:
+    path: "{{ cifmw_tobiko_artifacts_basedir }}"
+    state: directory
+
+- name: Configure tobiko
+  ansible.builtin.include_tasks: configure-tobiko.yml
+  when: not cifmw_tobiko_dry_run | bool
+
+- name: Copy CA bundle to cifmw_tobiko_artifacts_basedir
+  ansible.builtin.copy:
+    src: "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
+    dest: "{{ cifmw_tobiko_artifacts_basedir }}"
+    mode: '0444'
+    owner: "{{ lookup('env', 'USER') }}"
+    group: "{{ lookup('env', 'USER') }}"
+    remote_src: true
+
+- name: Set proper permission for tobiko directory
+  ansible.builtin.command:
+    cmd: "podman unshare chown 42495:42495 -R {{ cifmw_tobiko_artifacts_basedir }}"
+  when: not cifmw_tobiko_dry_run | bool
+
+- name: Ensure we have tobiko container image
+  register: _tobiko_fetch_img
+  containers.podman.podman_image:
+    name: "{{ cifmw_tobiko_image }}:{{ cifmw_tobiko_image_tag }}"
+  retries: 5
+  delay: 5
+  until: _tobiko_fetch_img is success
+
+- name: Run tobiko
+  ignore_errors: true
+  containers.podman.podman_container:
+    name: tobiko
+    image: "{{ cifmw_tobiko_image }}:{{ cifmw_tobiko_image_tag }}"
+    state: started
+    auto_remove: "{{ cifmw_tobiko_remove_container }}"
+    network: host
+    volume:
+      - "{{ cifmw_tobiko_artifacts_basedir }}/:/var/lib/tobiko/external_files:Z"
+      - "{{ cifmw_tobiko_artifacts_basedir }}/tls-ca-bundle.pem:/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem:Z"
+    detach: false
+    dns: "{{ cifmw_tobiko_dns_servers }}"
+    env:
+      TOBIKO_DEBUG: "{{ ci_tobiko_debug }}"
+      TOBIKO_TESTENV: "{{ ci_tobiko_testenv }}"
+      TOBIKO_PREVENT_CREATE: "{{ ci_tobiko_prevent_create if ci_tobiko_prevent_create is not none else omit }}"
```

Comment:
Omit is not going to work here I think. The value of the variable will be the "omit_<HASH>" calculated randomly by Ansible when starting. omit works only for module arguments.

Reply:
@pablintino it seems it works.
When I access the container, only the TOBIKO_* that are not "omitted" have been set.
```
[tobiko@controller-0 ~]$ env | grep ^TOBIKO
TOBIKO_TESTENV=neutron
TOBIKO_DEBUG=True
TOBIKO_UBUNTU_MINIMAL_IMAGE_URL=https://cloud-images.ubuntu.com/minimal/releases/jammy/release/ubuntu-22.04-minimal-cloudimg-amd64.img

```